### ISSUE TITLE
Improve GenerateEverything

### ISSUE THREAD

**Issue by MatthewDaggitt** [agda/agda-stdlib#1816]
Instead of hard coding lists of modules that use certain flags, read the files themselves and detecting the flags.

**Comment by gallais** [comment]
Hard-coding the list means that if we *forget* to set the right options then we get told off.
This helps ensure we do respect the invariants.

If we're happily inferring the file's status from its options, then we lose this benefit.

**Comment by jamesmckinna** [comment]
I got caught by this today though, (adding a file and failing to update the hard-coded lists) and it took me a while to figure out the source of the problem.

**Comment by jamesmckinna** [comment]
I'd favour closing thus issue, but not sure where is the best place in our documentation to summarise the principle here?

**Comment by JacquesCarette** [comment]
`HACKING.md` ?

**Comment by jamesmckinna** [comment]
> `HACKING.md` ?

Yes!. There was already text there, it turns out, but I've embellished it a little. #3063
